engine/class_modules/sc_hunter.cpp

```diff
@@ -983,6 +988,27 @@ struct hunter_action_t: public Base
     return total_regen + floor( total_energize );
   }
 
+  void consume_resource() override
+  {
+    ab::consume_resource();
+
+    if ( !p() -> options.t28_4pc )
+      return;
+
+    resource_e cr = ab::current_resource();
+    if ( cr != RESOURCE_FOCUS )
+      return;
+
+    p() -> state.focus_used += cost();
+
+    // TODO: The cost is a double, can this cause rounding errors?
+    while ( p() -> state.focus_used >= 80.0 )
+    {
+      p() -> state.focus_used -= 80.0;
+      p() -> buffs.trick_shots -> trigger( 2 );
+    }
+  }
+
```

Comment:
Tbh, I'd not bother implementing 4pc for now considering it simply doesn't make sense with the 2 charges wording.

Reply:
I'll leave it in for now

Reply:
Ok, then I'd move it to `hunter_t::resource_loss()`.

Reply:
Updated, please check it's implemented correct